pkg/asset/rhcos/image.go

```diff
@@ -140,14 +140,8 @@ func osImage(config *types.InstallConfig) (string, error) {
 		if oi := config.Platform.BareMetal.ClusterOSImage; oi != "" {
 			return oi, nil
 		}
-
-		// Note that baremetal IPI currently uses the OpenStack image
-		// because this contains the necessary ironic config drive
-		// ignition support, which isn't enabled in the UPI BM images
-		if a, ok := streamArch.Artifacts["openstack"]; ok {
-			return rhcos.FindArtifactURL(a)
-		}
-		return "", fmt.Errorf("%s: No openstack build found", st.FormatPrefix(archName))
+		// Use image from release payload
+		return "", nil
```

Comment:
Yes, I failed to notice that it could be collapsed, although the comment is perhaps helpful to make explicit what is going on.

If the user provides a value here we'll still pass it to the CBO in the Provisioning resource, and CBO will still ensure it is downloaded and cached. Default MachineSets won't refer to it any more but the user could still create a MachineSet that used the image they specified here and it would work.

We just don't default it to the OpenStack QCOW any more because we don't need that image by default; the new default is not to download any image.

Reply:
I agree that the comments are helpful and should be retained. If it is easier to describe the behavior by keeping the two separate returns, then that is perfectly acceptable.